src/main/java/org/beanpod/switchboard/dao/StreamDaoImpl.java

```diff
@@ -58,4 +71,25 @@ public void updateStream(StreamDTO streamDto) {
     StreamEntity streamEntity = mapper.toEntity(streamDto);
     streamRepository.save(streamEntity);
   }
+
+  private boolean onSamePrivateNetwork(
+      InputChannelDTO inputChannelDto, OutputChannelDTO outputChannelDto) {
+    return inputChannelDto
+        .getDecoder()
+        .getDevice()
+        .getPublicIpAddress()
+        .equals(outputChannelDto.getEncoder().getDevice().getPublicIpAddress());
+  }
+
+  private boolean deviceOnLocalNetwork(DeviceDTO deviceDto) {
+    return deviceDto.getPublicIpAddress().equals(deviceDto.getPrivateIpAddress())
+        || deviceDto.getPublicIpAddress().equals(LOOPBACK_IP_V4)
+        || deviceDto.getPublicIpAddress().equals(LOOPBACK_IP_V6);
```

Comment:
`DeviceDTO::publicIpAddress` is not actually the public IP address of the device, but the IP address of the device as seen from the server. If the device is on the same network as the server, this will not take on the value of the public IP, but the value of the local IP.

Granted, this only comes into play for local testing scenarios, so we can probably leave the name. But a comment explaining why it's possible for the public IP to be localhost could be useful I think. Also, good job thinking of the ipv6 case.

Reply:
Would you rather have publicIpAddress name requestIpAddress because in reality it's the request that we grab from the HTTP request.

Reply:
I don't think that helps much, let's stick with publicIpAddress.